internal/provider/resource_api_key.go

```diff
@@ -221,6 +227,7 @@ func mapToAPIKey(d *schema.ResourceData) search.Key {
 		MaxQueriesPerIPPerHour: d.Get("max_queries_per_ip_per_hour").(int),
 		Indexes:                castStringSet(d.Get("indexes")),
 		Referers:               castStringSet(d.Get("referers")),
+		QueryParameters:        d.Get("query_parameters").(string),
```

Comment:
This is also type error.  We would need to decode the string to `QueryParameters` struct like below.
https://github.com/algolia/algoliasearch-client-go/blob/40b333da0e6c734924e9ec40a831c63ad87f78bf/algolia/search/key.go#L70-L73